### ISSUE TITLE
I want to use AAC L2 codec in A2DP sink.

### ISSUE THREAD

**Issue by hyojincha** [espressif/esp-idf#12114]
The board I use consists of ESP32 + VS1053.

And I know that VS1053 itself can decode AAC.

Currently, Blueroid A2DP library seems to be fixed as SBC codec.

Is there any way to change this to AAC and receive undecoded sound source data through a callback function?

Please give me a solution.

Thank you. sir.

**Comment by boblane1** [comment]
Hi @hyojincha, 

Thanks for your report.  There is some other work to support AAC, such as A2DP stream discovering and codec negotiation. We intend to support other codec in the future, we lack human resource currently.

**Comment by boblane1** [comment]
Hi @hyojincha,

We accept your feature request, but regret to inform you that we are still short of human resources at this time. We may develop this feature later this year or next year.

**Comment by hyzgit** [comment]
> Hi [@hyojincha](https://github.com/hyojincha),  嗨，
> 
> We accept your feature request, but regret to inform you that we are still short of human resources at this time. We may develop this feature later this year or next year.我们接受您的功能请求，但很遗憾告知您，我们目前仍然缺乏人力资源。我们可能会在今年晚些时候或明年开发这个功能。

How is the progress of supporting Bluetooth AAC encoding at present?

**Comment by Finney-esp** [comment]
Hi, I'm very glad to tell you the following information：
- Currently, the master branch of ESP-IDF already supports A2DP sink using the AAC codec (external codec). The configuration to enable `BT_A2DP_CODEC_AAC_ENABLED` must be enabled, and the reported data is raw aac. The AAC decoding can be directly performed using `esp_audio_codec`.
- A2DP source currently does not support the AAC codec. Support for it will be provided in the near future. At the same time, the example in IDF will be improved in the near future to demonstrate how to use the AAC codec.

**Comment by hyzgit** [comment]
> Hi, I'm very glad to tell you the following information：嗨，我很高兴告诉您以下信息：
> 
> * Currently, the master branch of ESP-IDF already supports A2DP sink using the AAC codec (external codec). The configuration to enable `BT_A2DP_CODEC_AAC_ENABLED` must be enabled, and the reported data is raw aac. The AAC decoding can be directly performed using `esp_audio_codec`.目前，ESP-IDF 的主分支已经支持使用 AAC 编解码器（外部编解码器）的 A2DP 接收端。启用 `BT_A2DP_CODEC_AAC_ENABLED` 的配置必须被启用，报告的数据是原始 AAC 数据。可以使用 `esp_audio_codec` 直接进行 AAC 解码。
> * A2DP source currently does not support the AAC codec. Support for it will be provided in the near future. At the same time, the example in IDF will be improved in the near future to demonstrate how to use the AAC codec.当前的 A2DP 源端不支持 AAC 编解码器。它将在不久的将来提供支持。同时，IDF 中的示例将在不久的将来得到改进，以展示如何使用 AAC 编解码器。

Can it be connected via Bluetooth on the mobile phone and play music encoded in AAC? Will the mobile phone automatically recognize the device (ESP32) as a device that supports AAC?

**Comment by Finney-esp** [comment]
> > Hi, I'm very glad to tell you the following information：嗨，我很高兴告诉您以下信息：
> > 
> > * Currently, the master branch of ESP-IDF already supports A2DP sink using the AAC codec (external codec). The configuration to enable `BT_A2DP_CODEC_AAC_ENABLED` must be enabled, and the reported data is raw aac. The AAC decoding can be directly performed using `esp_audio_codec`.目前，ESP-IDF 的主分支已经支持使用 AAC 编解码器（外部编解码器）的 A2DP 接收端。启用 `BT_A2DP_CODEC_AAC_ENABLED` 的配置必须被启用，报告的数据是原始 AAC 数据。可以使用 `esp_audio_codec` 直接进行 AAC 解码。
> > * A2DP source currently does not support the AAC codec. Support for it will be provided in the near future. At the same time, the example in IDF will be improved in the near future to demonstrate how to use the AAC codec.当前的 A2DP 源端不支持 AAC 编解码器。它将在不久的将来提供支持。同时，IDF 中的示例将在不久的将来得到改进，以展示如何使用 AAC 编解码器。
> 
> Can it be connected via Bluetooth on the mobile phone and play music encoded in AAC? Will the mobile phone automatically recognize the device (ESP32) as a device that supports AAC?

Yes, it can be. If you want to use AAC codec in A2DP sink (ESP32), you should register a SEP whose codec type is `ESP_A2D_MCT_M24`. And then the mobile phone can automatically recognize the ESP32 as a device that supports AAC.
